numpy/_core/src/multiarray/multiarraymodule.c

```diff
@@ -3252,8 +3252,9 @@ array_set_datetimeparse_function(PyObject *NPY_UNUSED(self),
 NPY_NO_EXPORT PyObject *
 PyArray_Where(PyObject *condition, PyObject *x, PyObject *y)
 {
-    PyArrayObject *arr, *ax, *ay;
+    PyArrayObject *arr, *ax, *ay = NULL;
```

Comment:
Might as well initialize all of these to `NULL`, leaving it like this is a little confusing

Reply:
Added, although I can't say I prefer it when it isn't needed for cleanup (the "uninitialized" warnings work pretty well anyway).